### PR TITLE
BF: run fslpython_install.sh also for 6.0.1, also assure deprecation v1 version for 6.0.1

### LINKED ISSUE #271: Missing support python module(s)/libraries for fsleyes in fsl 6.0.1
Image was generated using
```bash
docker run -it --rm kaczmarj/neurodocker:master \
   generate singularity \
	--base=neurodebian:stretch \
	--fsl version=6.0.1 \
       -p apt \
	| sed  -e 's,/tmp/\* /var/tmp/\*,,g' \
	>| Singularity
```

env var for USER was adjusted per #270 by a call `source ~/set_user ` below.

```shell
$> singularity run -e -H /tmp/home ./fslanal
Some packages in this Docker container are non-free
If you are considering commercial use of this container, please consult the relevant license:
https://fsl.fmrib.ox.ac.uk/fsl/fslwiki/Licence
yoh@hopa:/home/yoh/proj/repronim/neurodocker-cases$ source ~/set_user 
yoh@hopa:/home/yoh/proj/repronim/neurodocker-cases$ DISPLAY=:0 fsleyes
Could not import wx
Traceback (most recent call last):
  File "fsl/utils/platform.py", line 159, in __init__
  File "/build.venv/lib/python3.6/site-packages/PyInstaller/loader/pyimod03_importers.py", line 631, in exec_module
  File "wxPython-4.0.3-py3.6-linux-x86_64.egg/wx/__init__.py", line 17, in <module>
  File "/build.venv/lib/python3.6/site-packages/PyInstaller/loader/pyimod03_importers.py", line 631, in exec_module
  File "wxPython-4.0.3-py3.6-linux-x86_64.egg/wx/core.py", line 12, in <module>
  File "/build.venv/lib/python3.6/site-packages/PyInstaller/loader/pyimod03_importers.py", line 714, in load_module
ImportError: libgtk-x11-2.0.so.0: cannot open shared object file: No such file or directory
Traceback (most recent call last):
  File "entrypt.py", line 146, in <module>
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
  File "/build.venv/lib/python3.6/site-packages/PyInstaller/loader/pyimod03_importers.py", line 631, in exec_module
  File "fsleyes/__init__.py", line 208, in <module>
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
  File "/build.venv/lib/python3.6/site-packages/PyInstaller/loader/pyimod03_importers.py", line 631, in exec_module
  File "fsleyes/plugins.py", line 176, in <module>
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
  File "/build.venv/lib/python3.6/site-packages/PyInstaller/loader/pyimod03_importers.py", line 631, in exec_module
  File "fsleyes/views/viewpanel.py", line 16, in <module>
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
  File "/build.venv/lib/python3.6/site-packages/PyInstaller/loader/pyimod03_importers.py", line 631, in exec_module
  File "wxPython-4.0.3-py3.6-linux-x86_64.egg/wx/__init__.py", line 12, in <module>
AttributeError: module 'wx' has no attribute '__version__'
[7988] Failed to execute script entrypt
```
I think the last one is the culprit: `AttributeError: module 'wx' has no attribute '__version__'`, and likely it is "upstream" issue for 6.0.1 release, and I didn't find a workaround yet, smth similar here https://www.jiscmail.ac.uk/cgi-bin/webadmin?A2=FSL;f2962d60.1812 was not applicable since I do not see any `libfreetype*`.  May be @pauldmccarthy has an immediate idea?

### PR DESCRIPTION
**Pull request by yarikoptic**
Fixes #271 

TODOs
- [x] verify that it fixes it ;)

### CODE REVIEW

File: neurodocker/templates/fsl.yaml [hunk 1]
```diff
@@ -41,7 +41,11 @@ generic:
...
       sed -i '$isource $FSLDIR/etc/fslconf/fsl.sh' $ND_ENTRYPOINT
-      {% if fsl.version in ("5.0.11", "5.0.10") -%}
+      {% if fsl.version in ("5.0.11", "5.0.10", "6.0.1") -%}
       echo "Installing FSL conda environment ..."
       bash {{ fsl.install_path }}/etc/fslconf/fslpython_install.sh -f {{ fsl.install_path }}
       {% endif -%}
+      {% if fsl.version in ("6.0.1") -%}
+      echo "Downgrading deprecation module per https://github.com/kaczmarj/neurodocker/issues/271#issuecomment-514523420"
+      bash {{ fsl.install_path }}/fslpython/bin/conda install -n fslpython -c conda-forge -y deprecation==1.*
+      {% endif -%}
```
Comment (yarikoptic) [inline neurodocker/templates/fsl.yaml]: that was wrong, probably just need to remove `bash ` for this invocation ;)
```shell
+ sed -i $iecho If you are considering commercial use of this container, please consult the relevant license: /neurodocker/startup.sh
+ sed -i $iecho https://fsl.fmrib.ox.ac.uk/fsl/fslwiki/Licence /neurodocker/startup.sh
+ sed -i $isource $FSLDIR/etc/fslconf/fsl.sh /neurodocker/startup.sh
+ echo Installing FSL conda environment ...
Installing FSL conda environment ...
+ bash /opt/fsl-6.0.1/etc/fslconf/fslpython_install.sh -f /opt/fsl-6.0.1
Stage 1
By installing this python distribution you agree to the license terms in
/opt/fsl-6.0.1/fslpython/LICENSE.txt
100%
Stage 2
100%
/tmp/fslpythonMABD/fslpython_miniconda_installer.log
+ echo Downgrading deprecation module per https://github.com/kaczmarj/neurodocker/issues/271#issuecomment-514523420
Downgrading deprecation module per https://github.com/kaczmarj/neurodocker/issues/271#issuecomment-514523420
+ bash /opt/fsl-6.0.1/fslpython/bin/conda install -n fslpython -c conda-forge -y deprecation==1.*
/opt/fsl-6.0.1/fslpython/bin/conda: line 3: import: command not found
/opt/fsl-6.0.1/fslpython/bin/conda: line 6: syntax error near unexpected token `sys.argv'
/opt/fsl-6.0.1/fslpython/bin/conda: line 6: `if len(sys.argv) > 1 and sys.argv[1].startswith('shell.') and sys.path and sys.path[0] == '':'
```

**Comment by yarikoptic** [comment]
it seems to resolve the issue of having stuff not installed BUT at the cost of pretty much doubling (3.2GB for fslpython) the size of the image!  Here is the heaviest added parts:

```
--- /opt/fsl-6.0.1/fslpython/pkgs --
  747.2 MiB [##########] /mkl-2019.4-243
  287.1 MiB [###       ] /qt-5.9.7-h5867ecd_1
  204.1 MiB [##        ]  mkl-2019.4-243.tar.bz2
  177.6 MiB [##        ] /vtk-8.2.0-py37haa4764d_200
  115.4 MiB [#         ] /cache                                                                                                                                                                                                   
   85.9 MiB [#         ]  qt-5.9.7-h5867ecd_1.tar.bz2
   84.5 MiB [#         ] /wxpython-4.0.4-py37hc99224d_0
   83.7 MiB [#         ] /python-3.7.3-h0371630_0
...
```
I wonder if `mkl` and cache are needed for the run time at all, or what could be stripped/removed to @pauldmccarthy

**Comment by yarikoptic** [comment]
here is the image I generated if anyone interested to test: http://datasets.datalad.org/singularity/test-fsl-6.0.1-neurodocker.sing  (it is not under git-annex there despite the website ;))

**Comment by pauldmccarthy** [comment]
Aah; unfortunately mkl is required, as FSL now uses armadillo (which uses LAPACK/BLAS under the hood) as its linear algebra library.

Perhaps in your case it is worth using a modified `fslpython_environnment.yml` which omits all of the unnecessary libraries (e.g. `qt`, `wxpython`, `vtk` etc); `qt` and `vtk` are only used by MIST (sub-cortical segmentation), and `wxpythin` is only used by FSLeyes.

**Comment by yarikoptic** [comment]
hm, I could have sworn that I have posted the follow up but I don't see it.

In that new environment I am out of luck running fsleyes. @pauldmccarthy - do you have a linux env with singularity?  could you checkout that  http://datasets.datalad.org/singularity/test-fsl-6.0.1-neurodocker.sing to see what is missing/wrong?

**Comment by yarikoptic** [comment]
I just verified that created container is at least working to run 2nd level analysis (i.e. `imglob` is there and didn't freak out).  So I think it would be good to have this PR merged and then improve later (after all the fsleyes issues are tracked down)

**Comment by pauldmccarthy** [comment]
@yarikoptic I've got it running - looks like a zlib version incompatibility. I'll have a play tomorrow, and will get back to you

**Comment by kaczmarj** [comment]
thanks @yarikoptic and @pauldmccarthy - i will test out this pr and merge

what is fslpython necessary for? is it something that could be optionally installed with a command line flag?

**Comment by yarikoptic** [comment]
I guess it is all python based pieces of fsl. Some are command line tools used eg for 2nd level analysis,

**Comment by pauldmccarthy** [comment]
Sorry for the delay; I left a `singularity build` command running for most of yesterday to convert your image into a sandbox directory, only to have my VM run out of hard drive space. I might have to leave this until the weekend or next week, as I'm going to be away for a couple of days.

`fslpython` is a miniconda environment which gets installed into `$FSLDIR` during the installation process. It basically gives the FSL team an environment to run Python-based tools, and was added to assist in the slow migration of the FSL code base away from bash and TCL/TK.

Currently it is not used by a large number of tools, but there are a few critical ones that have been ported to use it (e.g. `imcp`, `imglob`, `atlasquery` etc). It is also used by `bianca` and `mist`; In the next release, FSLeyes will also be installed directly into the `fslpython` environment, rather than being built independently. And there are likely to be more Python-based tools (and ports of existing tools) in the future. So unfortunately it's not optional.

But you could trim it down; for example, if you don't care about `mist`, you can remove `qt` and `vtk`; if you don't care about FSLeyes, you can remove `wxpython`, etc etc.

**Comment by yarikoptic** [comment]
> I left a `singularity build` command running for most of yesterday to convert your image into a sandbox directory, only to have my VM run out of hard drive space

why bother and not just run it? ;)

**Comment by pauldmccarthy** [comment]
> why bother and not just run it? ;)

I wanted to `yum install` some extra stuff to try and fix the FSLeyes problem, but I was told that I was working with a read only file system. I thought that only sandbox images were writeable, but it's been a while since I've used singularity. Anyway, I won't be able to touch this until next week unfortunately... will get back to you when I get a chance.

**Comment by yarikoptic** [comment]
I think you can add an overlay image. Iirc it is Debian based, so use apt ;-)

**Comment by pauldmccarthy** [comment]
Aah, thanks for the tip!

>  Iirc it is Debian based

Duh; it's already been a couple of days since I looked at it :)

**Comment by kaczmarj** [comment]
Thanks @yarikoptic and @pauldmccarthy.

**Comment by pauldmccarthy** [comment]
Finally got a chance to play with this. There are two problems:

 - There seems to be an incompatibility in the version of `libz` that is bundled with FSLeyes - removing `/opt/fsl-6.0.1/bin/FSLeyes/libz.so.1` fixes this problem.

 - Some mesa libs seem to be missing - installing `libgl1-mesa-dri` fixes this problem.

After addressing both of these problems, FSLeyes seems to work fine. The first problem should become obsolete in the next version of FSL, as FSLeyes will be installed via conda, instead of being built separately.

**Comment by yarikoptic** [comment]
Thanks @pauldmccarthy !  I am testing those out and will submit PR if it works for me ;) Cheers!

**Pull request merged**